fx quant: clean up nit in insert_observer

vkuzo opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #57402 fx quant: remove `find_quants` from convert
* #57399 fx quant: remove unnecessary quants arguments
* #57393 fx quant: remove FixedQParamsOpQuantizeHandler from quantize.py
* #57388 fx quant: move input_output_observed to qhandler
* #57377 fx quant: move output obs logic to QuantizeHandler
* #57368 fx quant: readability improvements on observer functions
* **#57367 fx quant: clean up nit in insert_observer**

Summary:

This code is never hit (see insert_observer_for_output_of_the_node
which gates it out), so changing to an assert in order to
have `insert_observer` actually always insert an observer.
This helps code readability.

Test Plan:

```
python test/test_quantization.py TestQuantizeFx
```

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D28126898](https://our.internmc.facebook.com/intern/diff/D28126898)

vkuzo committed:
fx quant: clean up nit in insert_observer

Summary:

This code is never hit (see insert_observer_for_output_of_the_node
which gates it out), so changing to an assert in order to
have `insert_observer` actually always insert an observer.
This helps code readability.

Test Plan:

```
python test/test_quantization.py TestQuantizeFx
```

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 commented on torch/quantization/fx/quantize.py:
```diff
@@ -116,8 +116,9 @@ def insert_observer(
     # won't insert observer for them
-    if not model.training and isinstance(observer, torch.quantization.FixedQParamsFakeQuantize):
-        return
+    if not model.training:
```
maybe we can remove this then? it's probably added before these checks?

vkuzo replied:
this assert was actually pretty useful in the PRs further in the stack, it helped iterate on the code with useful error messages, maybe we can keep this?

jerryzh168 replied:
I see, OK

vkuzo committed:
Update on "fx quant: clean up nit in insert_observer"


Summary:

This code is never hit (see insert_observer_for_output_of_the_node
which gates it out), so changing to an assert in order to
have `insert_observer` actually always insert an observer.
This helps code readability.

Test Plan:

```
python test/test_quantization.py TestQuantizeFx
```

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D28126898](https://our.internmc.facebook.com/intern/diff/D28126898)

[ghstack-poisoned]
